plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryParentProjectId.java

```diff
@@ -0,0 +1,116 @@
+                .isEqualTo("ALGERIA");
+        assertThat(computeScalar(format("SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT name FROM %s.tpch.nation WHERE nationkey = 0'))", testingParentProjectId)))
+                .isEqualTo("ALGERIA");
+
+        String trinoSchema = "someschema_" + randomNameSuffix();
+        try (AutoCloseable ignored = withSchema(trinoSchema); TestTable table = newTrinoTable("%s.test_create_table_already_exists".formatted(trinoSchema), "(col1 INT)")) {
+            String tableName = table.getName().split("\\.")[1];
+            // schema created in testingProjectId is present in testingProjectId and NOT present in testingParentProjectId
+            assertThat(computeActual(format(
+                    "SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT schema_name FROM `%s.region-us.INFORMATION_SCHEMA.SCHEMATA`'))",
+                    testingProjectId)))
+                    .contains(new MaterializedRow(DEFAULT_PRECISION, trinoSchema));
+            // confusion point: this implicitly points to parent project
+            assertThat(computeActual(format("SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT schema_name FROM INFORMATION_SCHEMA.SCHEMATA'))")))
+                    .doesNotContain(new MaterializedRow(DEFAULT_PRECISION, trinoSchema));
+            assertThat(computeActual(format(
+                    "SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT schema_name FROM `%s.region-us.INFORMATION_SCHEMA.SCHEMATA`'))",
+                    testingParentProjectId)))
+                    .doesNotContain(new MaterializedRow(DEFAULT_PRECISION, trinoSchema));
+            // table created in testingProjectId is present in testingProjectId and NOT present in testingParentProjectId
+            assertThat(computeActual("SHOW TABLES FROM " + trinoSchema).getOnlyColumn()).contains(tableName);
+            assertThat(query(format(
+                    "SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT table_name FROM INFORMATION_SCHEMA.TABLES WHERE table_schema = \"%s\"'))",
+                    trinoSchema)))
+                    .failure()
+                    .hasMessageContaining("Table \"INFORMATION_SCHEMA.TABLES\" must be qualified with a dataset (e.g. dataset.table)");
+            assertThat(computeActual(format(
+                    "SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT table_name FROM `%s.region-us.INFORMATION_SCHEMA.TABLES` WHERE table_schema = \"%s\"'))",
+                    testingParentProjectId,
+                    trinoSchema)))
+                    .doesNotContain(row(tableName));
+            assertThat(query("SELECT * FROM " + table.getName())).returnsEmptyResult();
+            assertThat(query("SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT * FROM `%s.%s`'))".formatted(testingParentProjectId, table.getName())))
```

Comment:
Can you add similar query with explicit `testingProjectId`?

Reply:
Done!